test/OpenIddict.Abstractions.Tests/Primitives/OpenIddictConverterTests.cs

```diff
@@ -0,0 +1,322 @@
+        [InlineData(typeof(OpenIddictResponse[]), false)]
+        [InlineData(typeof(OpenIddictParameter), false)]
+        [InlineData(typeof(OpenIddictParameter?), false)]
+        [InlineData(typeof(OpenIddictParameter[]), false)]
+        [InlineData(typeof(OpenIddictParameter?[]), false)]
+        [InlineData(typeof(object), false)]
+        [InlineData(typeof(long), false)]
+        public void CanConvert_ReturnsExpectedResult(Type type, bool result)
+        {
+            // Arrange
+            var converter = new OpenIddictConverter();
+
+            // Act and assert
+            Assert.Equal(result, converter.CanConvert(type));
+        }
+
+        [Fact]
+        public void ReadJson_ThrowsAnExceptionForNullReader()
+        {
+            // Arrange
+            var converter = new OpenIddictConverter();
+
+            // Act and assert
+            var exception = Assert.Throws<ArgumentNullException>(() =>
+            {
+                converter.ReadJson(reader: null, type: null, value: null, serializer: null);
+            });
+
+            Assert.Equal("reader", exception.ParamName);
+        }
+
+        [Fact]
+        public void ReadJson_ThrowsAnExceptionForNullType()
+        {
+            // Arrange
+            var converter = new OpenIddictConverter();
+
+            // Act and assert
+            var exception = Assert.Throws<ArgumentNullException>(() =>
+            {
+                converter.ReadJson(reader: new JsonTextReader(TextReader.Null), type: null, value: null, serializer: null);
+            });
+
+            Assert.Equal("type", exception.ParamName);
+        }
+
+        [Fact]
+        public void ReadJson_ThrowsAnExceptionForUnexpectedJsonToken()
+        {
+            // Arrange
+            var converter = new OpenIddictConverter();
+            using var reader = new JsonTextReader(new StringReader("[0,1,2,3]"));
+
+            // Act and assert
+            var exception = Assert.Throws<JsonSerializationException>(() =>
+            {
+                converter.ReadJson(reader: reader, type: typeof(OpenIddictRequest), value: null, serializer: null);
+            });
+
+            Assert.Equal("An error occurred while reading the JSON payload.", exception.Message);
+        }
+
+        [Theory]
+        [InlineData(typeof(OpenIddictMessage[]))]
+        [InlineData(typeof(OpenIddictRequest[]))]
+        [InlineData(typeof(OpenIddictResponse[]))]
+        [InlineData(typeof(OpenIddictParameter))]
+        [InlineData(typeof(OpenIddictParameter?))]
+        [InlineData(typeof(OpenIddictParameter[]))]
+        [InlineData(typeof(OpenIddictParameter?[]))]
+        [InlineData(typeof(object))]
+        [InlineData(typeof(long))]
+        public void ReadJson_ThrowsAnExceptionForUnsupportedType(Type type)
+        {
+            // Arrange
+            var converter = new OpenIddictConverter();
+            using var reader = new JsonTextReader(new StringReader(@"{""name"":""value""}"));
```

Comment:
I like that. Should we add `using` in the other tests instantiating `JsonTextReader`? (at least for consistency)

Reply:
I don't think it's necessary right now. I faced only 2 places with warnings (which I fixed)